packages/terra-slider/src/Slider.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import ThemeContext from 'terra-theme-context';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
+import { v4 as uuidv4 } from 'uuid';
+import styles from './Slider.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+  * The label for the slider.
+  */
+  label: PropTypes.string.isRequired,
+
+  /**
+  * Minimum value in the range.
+  */
+  minimumValue: PropTypes.number.isRequired,
+
+  /**
+  * Maximum value in the range.
+  */
+  maximumValue: PropTypes.number.isRequired,
+
+  /**
+  * Default or initial value for the slider.
+  */
+  value: PropTypes.number.isRequired,
+
+  /**
+  * When true, will disable the field.
+  */
+  isDisabled: PropTypes.bool,
+
+  /**
+  * Minimum value label to display on the left side of the slider.
+  */
+  minimumLabel: PropTypes.string,
+
+  /**
+  * Maximum value label to display on the right side of the slider.
+  */
+  maximumLabel: PropTypes.string,
+
+  /**
+  * Function to trigger when user changes the input value. Sends parameter {Event} event.
+  */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  isDisabled: false,
+  onChange: null,
+};
+
+class Slider extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: this.props.value,
+    };
+    this.handleOnChange = this.handleOnChange.bind(this);
+    this.sliderRef = React.createRef();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.value === prevProps.value) {
+      return;
+    }
+    // eslint-disable-next-line react/no-did-update-set-state
+    this.setState({
+      value: this.props.value,
+    });
+  }
+
+  handleOnChange(event) {
+    this.setState({ value: event.currentTarget.value });
```

Comment:
If the value has already been changed in the component, why do we need to set state?